src/sentry/tasks/store.py

```diff
@@ -562,11 +562,18 @@ def _do_save_event(
             data = manager.get_data()
             if not isinstance(data, dict):
                 data = dict(data.items())
-            processing.event_processing_store.store(data)
+
+            if event_type == "transaction":
+                processing.transaction_processing_store.store(data)
+            else:
+                processing.event_processing_store.store(data)
         except HashDiscarded:
             # Delete the event payload from cache since it won't show up in post-processing.
             if cache_key:
-                processing.event_processing_store.delete_by_key(cache_key)
+                if event_type == "transaction":
+                    processing.transaction_processing_store.delete_by_key(cache_key)
+                else:
+                    processing.event_processing_store.delete_by_key(cache_key)
```

Comment:
you have this if/else block twice in this file..

suggest to refactor to something like:
```
processing_store = transaction_processing_store if event_type == "transaction" else event_processing_store
processing_store.store(data)
processing_store.delete_by_key(cache_key)
```

also, there needs to be a 3rd time, as line 509 also needs to go to either the event or transaction processing store